Tasks/VsTest/settingshelper.ts

```diff
@@ -82,8 +82,8 @@ export async function updateSettingsFileAsRequired(settingsFile: string, isParal
     }
 
     if (overrideParametersString) {
-        if (settingsExt === runSettingsExtension) {
-            result = updateRunSettingsWithParameters(result, overrideParametersString);
+        if (settingsExt === runSettingsExtension || settingsExt === testSettingsExtension) {
+            result = updateSettingsWithParameters(result, overrideParametersString);
```

Comment:
I would recommend a "TestSettings" class and have associated properties and methods.. This "if" branching all over the place in this makes to difficult to read..
I know its not just specific to your change.. But we can this as an opportunity to make this code better

Reply:
Will not be refactoring with this change, as the all the related tests also will need to be refactored. Will open another issue to track this.

---
In reply to: [136552653](https://github.com/Microsoft/vsts-tasks/pull/5191#discussion_r136552653) [](ancestors = 136552653)